app/models/kubernetes_image_scan.rb

```diff
@@ -0,0 +1,264 @@
+#
+
+require 'kubeclient'
+require 'securerandom'
+
+class KubernetesImageScan < Job
+  PROVIDER_CLASS = ManageIQ::Providers::Kubernetes::ContainerManager
+  INSPECTOR_IMAGE = 'docker.io/fsimonce/image-inspector:v0.1.2'
+  INSPECTOR_NAMESPACE = 'default'
+  INSPECTOR_PORT = 8080
+  DOCKER_SOCKET = '/var/run/docker.sock'
+  SCAN_CATEGORIES = %w(system software)
+  POD_POLL_INTERVAL = 10
+  KUBERNETES_GUEST_OS = 'Linux'
+  INSPECTOR_HEALTH_PATH = '/healthz'
+  INSPECTOR_CONTENT_PATH = '/api/v1/content'
+  ERRCODE_POD_NOTFOUND = 404
+
+  def load_transitions
+    self.state ||= 'initializing'
+    {
+      :initializing => {'initializing'       => 'waiting_to_start'},
+      :start        => {'waiting_to_start'   => 'creating_pod'},
+      :wait_pod     => {'creating_pod'       => 'waiting_to_analyze'},
+      :analyze      => {'waiting_to_analyze' => 'analyzing'},
+      :cleanup      => {'analyzing'          => 'removing_pod'},
+      :abort_job    => {'*'                  => 'aborting'},
+      :finish       => {'removing_pod' => 'finished',
+                        'aborting'     => 'finished'},
+    }
+  end
+
+  def initializing
+    queue_signal(:start)
+  end
+
+  def start
+    image = target_entity
+    return queue_signal(:abort_job) unless image
+
+    update!(:options => options.merge(
+      :ems_id          => image.ext_management_system.id,
+      :scan_id         => SecureRandom.uuid,
+      :docker_image_id => image.docker_id,
+      :image_full_name => image.full_name,
+      :pod_name        => "manageiq-img-scan-#{image.docker_id[0..11]}",
+      :pod_port        => INSPECTOR_PORT,
+      :pod_namespace   => INSPECTOR_NAMESPACE
+    ))
+
+    pod = pod_definition
+    _log.info("creating pod #{pod_full_name} to analyze docker image " \
+              "#{options[:docker_image_id]}: #{pod.inspect}")
+
+    begin
+      kubernetes_client.create_pod(pod)
+    rescue KubeException => e
+      _log.info("pod creation for #{pod_full_name} failed: #{e}")
+      return queue_signal(:abort_job)
+    end
+
+    queue_signal(:wait_pod)
+  end
+
+  def wait_pod
+    _log.info("waiting for pod #{pod_full_name} to be available")
+
+    client       = kubernetes_client
+    health_url   = pod_proxy_url(client, INSPECTOR_HEALTH_PATH)
+    http_options = {
+      :use_ssl     => health_url.scheme == 'https',
+      :verify_mode => ext_management_system.verify_ssl_mode
+    }
+
+    loop do
+      response = pod_health_poll(client, health_url, http_options)
+
+      case response
+      when Net::HTTPOK
+        _log.info("pod #{pod_full_name} is ready and accessible")
+        break
+      when Net::HTTPServiceUnavailable
+        # continue: pod is still not up and running
+      else
+        _log.info("unknown access error to pod #{pod_full_name}: " \
+                  "#{response}")
+        return queue_signal(:abort_job)
+      end
+
+      _log.info("pod #{pod_full_name} is not available")
+      sleep(POD_POLL_INTERVAL)
```

Comment:
@gtanzillo according to your suggestions, do you think it would be better to wait using the queue (even possible?)?
I actually have mixed feelings about it (again if possible). We could try to discuss it together.